### PR TITLE
Stream server for exec, attach and port forward

### PR DESCRIPTION
**Pull request by sameo**
This is a fork of PR #282, with `ocic ctr exec`, tty resizing and vendoring updated.

### CODE REVIEW

**Comment by mrunalp** [comment]
Yay all green :) I will run e2e tests on this..

**Comment by mrunalp** [comment]
119/121 :)

```
Summarizing 2 Failures:

[Fail] [k8s.io] Projected [It] should project all components that make up the projection API [Conformance] [Volume] [Projection] 
/root/gosrc/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/util.go:2213

[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
/root/gosrc/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/exec_util.go:107

Ran 121 of 213 Specs in 3891.882 seconds
FAIL! -- 119 Passed | 2 Failed | 0 Pending | 92 Skipped --- FAIL: TestE2eNode (3891.89s)
```

**Comment by sameo** [comment]
@mrunalp Could:

```
[Fail] [k8s.io] KubeletManagedEtcHosts [It] should test kubelet managed /etc/hosts file [Conformance] 
```

be fixed with #468 ?

**Comment by mrunalp** [comment]
With #468 on top of this, we also pass that test :)

```
• [SLOW TEST:70.878 seconds]
[k8s.io] KubeletManagedEtcHosts
/root/gosrc/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/framework/framework.go:656
  should test kubelet managed /etc/hosts file [Conformance]
  /root/gosrc/src/k8s.io/kubernetes/_output/local/go/src/k8s.io/kubernetes/test/e2e/common/kubelet_etc_hosts.go:55
------------------------------
SSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSSI0421 19:39:48.140396     423 e2e_node_suite_test.go:169] Stopping node services...
I0421 19:39:48.140408     423 server.go:303] Kill server "services"
I0421 19:39:48.140436     423 server.go:340] Killing process 587 (services) with -TERM
I0421 19:39:48.172407     423 server.go:303] Kill server "kubelet"
I0421 19:39:48.179455     423 services.go:156] Fetching log files...
I0421 19:39:48.179568     423 services.go:165] Get log file "kern.log" with journalctl command [-k].
I0421 19:39:48.394073     423 services.go:165] Get log file "docker.log" with journalctl command [-u docker].
I0421 19:40:01.975325     423 services.go:165] Get log file "cloud-init.log" with journalctl command [-u cloud*].
E0421 19:40:02.901198     423 services.go:168] failed to get "cloud-init.log" from journald: Failed to add filter for units: No data available
, exit status 1
I0421 19:40:02.901899     423 services.go:165] Get log file "kubelet.log" with journalctl command [-u kubelet-1369752116.service].
I0421 19:40:02.951746     423 e2e_node_suite_test.go:174] Tests Finished

Ran 1 of 213 Specs in 89.892 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 212 Skipped PASS
```

**Comment by mrunalp** [comment]
LGTM

**Comment by runcom** [comment]
LGTM

**Comment by mrunalp** [comment]
@cyphar PTAL

File: cmd/ocic/container.go [hunk 1]
```diff
@@ -236,6 +242,45 @@ var execSyncCommand = cli.Command{
...
+	Usage: "prepare a streaming endpoint to execute a command in the container",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "id",
+			Value: "",
+			Usage: "id of the container",
+		},
+		cli.BoolFlag{
+			Name:  "tty",
+			Usage: "wheter to use tty",
```
Comment (TomSweeneyRedHat) [inline cmd/ocic/container.go]: wheter to whether here and through out.

File: cmd/ocic/container.go [hunk 2]
```diff
@@ -236,6 +242,45 @@ var execSyncCommand = cli.Command{
...
+		},
+		cli.BoolFlag{
+			Name:  "stdin",
+			Usage: "wheter to stream to stdin",
```
Comment (TomSweeneyRedHat) [inline cmd/ocic/container.go]: whether

**Comment by runcom** [comment]
Ping @cyphar

**Pull request merged**